libvast/vast/chunk.hpp

```diff
@@ -124,6 +90,20 @@ class chunk final : public caf::ref_counted {
   /// Avoid the common mistake of binding ownership to a string view.
   static auto make(std::string_view&&) = delete;
 
+  /// Construct a chunk from a byte buffer by copying it.
+  /// @param buffer The byte buffer.
+  /// @returns A chunk pointer or `nullptr` on failure.
+  template <class Buffer>
+  static auto copy(const Buffer& buffer)
```

Comment:
This works for now. Once our chunks support allocators, we could offer a second argument that contains an allocator.